index.js

```diff
@@ -3983,6 +3983,26 @@
   S.splitOn =
   def('splitOn', {}, [$.String, $.String, $.Array($.String)], splitOn);
 
+  //# splitOnRegex :: RegExp -> String -> Array String
+  //.
+  //. Returns the substrings of its second argument separated by evaluated
+  //. matches of its first argument.
```

Comment:
I like @cust0dian's suggestion, though it could be misinterpreted. Given `/x/` and `'abc'`, the substrings that do not match the pattern are `'abc'`, `'ab'`, `'bc'`, `'a'`, `'b'`, `'c'`, and `''` (perhaps).

Another option:

_Takes a pattern and a string, and returns the result of splitting the string at every non-overlapping occurrence of the pattern (removing the matched substrings)._